src/Utilities/ErrorHandling/Strerror.cpp

```diff
@@ -0,0 +1,17 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Utilities/ErrorHandling/Strerror.hpp"
+
+#include <array>
+#include <string>
+
+#include "Utilities/ErrorHandling/StrerrorWrapper.h"
+
+std::string strerror_threadsafe(const int errnum) {
+  std::array<char, 1000> message{};
```

Comment:
Could you add a comment (when you squash) about why 1000 is enough? I'm worried about overflow, but that worry might be misguided.